gosub-bindings/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::os::raw::c_char;
+use std::ptr;
+
+use gosub_engine::{
+    bytes::{CharIterator, Confidence, Encoding},
+    html5::parser::{
+        document::{Document, DocumentBuilder},
+        Html5Parser,
+    },
+    render_tree::{Node, NodeType, RenderTree, TreeIterator},
+};
+
+#[no_mangle]
+/// Initialize a render tree and return an owning pointer to it.
+/// If the HTML fails to parse, returns a NULL pointer.
+///
+/// # Safety
+/// Takes a read-only pointer owned from the C API representing the HTML source
+/// to build a render tree. DO NOT take ownership of this pointer in Rust or the
+/// universe might collapse.
+///
+/// Moves an owning pointer to the render_tree using Box::into_raw() to the C API.
+/// This pointer MUST be passed to gosub_render_tree_free() after usage for proper cleanup.
+pub unsafe extern "C" fn gosub_render_tree_init(html: *const c_char) -> *mut RenderTree {
+    let html_str: &str;
+    unsafe {
+        html_str = CStr::from_ptr(html).to_str().unwrap();
+    }
+    let mut chars = CharIterator::new();
+    chars.read_from_str(html_str, Some(Encoding::UTF8));
+    chars.set_confidence(Confidence::Certain);
+
+    let doc = DocumentBuilder::new_document();
+    let parse_result = Html5Parser::parse_document(&mut chars, Document::clone(&doc), None);
```

Comment:
I've noticed here, we maybe should not return a Result with a `Vec<ParseError>` as the `Ok` type. Maybe i missed something here?

Reply:
Not sure, @jaytaph would have more insight to that, I am just a consumer of the parser :P